tokio-tcp/src/stream.rs

```diff
@@ -563,6 +604,27 @@ impl<'a> futures2::io::AsyncWrite for &'a TcpStream {
         futures2::io::AsyncWrite::poll_write(&mut &self.io, cx, buf)
     }
 
+    fn poll_vectored_write(&mut self, cx: &mut futures2::task::Context, vec: &[&IoVec])
+        -> futures2::Poll<usize, io::Error>
+    {
+        if let futures2::Async::Pending = self.io.poll_write_ready2(cx)? {
+            return Ok(futures2::Async::Pending)
+        }
+
+        let r = self.io.get_ref().write_bufs(vec);
+
+        match r {
+            Ok(n) => {
+                Ok(futures2::Async::Ready(n))
+            }
+            Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
+                self.io.clear_write_ready()?;
```

Comment:
Seems this line should be `self.io.clear_write_ready2(cx)?;` I encountered panic "no Task is currently running", and finally shot this line.

Reply:
Woops, yes it should!